mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergSplit.java

```diff
@@ -74,9 +74,11 @@ public long getLength() {
 
   @Override
   public String[] getLocations() {
-    if (locations == null) {
+    if (locations == null && conf != null) {
```

Comment:
I think a comment can help here saying `implementation of getLocations is only meant to be used on the client side during splits computation.  getLocations won't be accurate when called on worker nodes `

Reply:
Done